tests/functionnal/Search.php

```diff
@@ -524,29 +503,44 @@ public function testSearchAllMeta() {
                $criterion_params['itemtype'] = $metaitemtype;
                $criterion_params['link'] = 'AND';
 
-               $metacriteria[] = $criterion_params;
+               $itemtype_criteria[$itemtype][] = $criterion_params;
+            }
+         }
+      }
 
-               // Search with each meta criteria independently.
-               $search_params = ['is_deleted'   => 0,
-                                 'start'        => 0,
-                                 'criteria'     => [0 => ['field'      => 'view',
-                                                          'searchtype' => 'contains',
-                                                          'value'      => '']],
-                                 'metacriteria' => [$criterion_params]];
-               $this->doSearch($itemtype, $search_params);
+      foreach ($itemtype_criteria as $itemtype => $criteria) {
+         if (empty($criteria)) {
+            continue;
+         }
+
+         $first_criteria_by_metatype = [];
+
+         // Search with each meta criteria independently.
+         foreach ($criteria as $criterion_params) {
+            if (!array_key_exists($criterion_params['itemtype'], $first_criteria_by_metatype)) {
+               $first_criteria_by_metatype[$criterion_params['itemtype']] = $criterion_params;
             }
+
+            $search_params = ['is_deleted'   => 0,
+                              'start'        => 0,
+                              'criteria'     => [0 => ['field'      => 'view',
+                                                       'searchtype' => 'contains',
+                                                       'value'      => '']],
+                              'metacriteria' => [$criterion_params]];
+            $this->doSearch($itemtype, $search_params);
          }
 
-         // Search on all with multiple meta criteria
-         // Limit criteria count to 50 to prevent performances issues
-         // and also prevent exceeding of MySQL join limit.
-         foreach (array_chunk($metacriteria, 50) as $metacriteria_chunk) {
+         // Search with criteria related to multiple meta items.
+         // Limit criteria count to 5 to prevent performances issues (mainly on MariaDB).
+         // Test would take hours if done using too many criteria on each request.
+         // Thus, using 5 different meta items on a request seems already more than a normal usage.
+         foreach (array_chunk($first_criteria_by_metatype, 5) as $criteria_chunk) {
```

Comment:
According to my tests, we could gain something like 30 seconds on MariaDB by reducing this limit to 3.

Reply:
Problem is you try to call one search query with all criteria, right? Indeed, this is not a use case.

I guess we could keep a maximum or 2 or 3 criteria; making sure each individual one is tested that way.

Reply:
For now, all criteria are tested independently (one search query for each criteria), then, in a second time, first criteria of each metatype are tested by groups (one search query with 5 criteria).

Reply:
Ok, so I guess limiting here to 3 would be enough on tests runs

Reply:
```suggestion
         foreach (array_chunk($first_criteria_by_metatype, 3) as $criteria_chunk) {
```